utils/auto_dockerfile_update/update_dockerfiles.py

```diff
@@ -40,13 +40,159 @@ def is_docker_file_outdated(dockerfile: Dict, latest_tag: str, last_updated: str
+    # if we get the docker image version it is only numeric so we want to add it ~ to pyproject.
+    # if we get numeric version it should be only for replace it in the pyproject after lock failure.
+    if is_numeric_version:
+        if "pyproject" in file_path:
+            return f"~{version_array[0]}.{version_array[1]}"     
+        return f"{version_array[0]}.{version_array[1]}"   
+    return version
+
+def replace_python_version(file_path: str, version: str,
+                           full_str_python_version:str)->bool:
+    """Replace the current python version in the Pipfile or the pyproject.toml.
+    
+        Args:
+            file_path (str): The file path to the Pipfile or the pyproject.toml file.
+            version (str): The updated/old version.
+            full_str_python_version (str): The older version.
+    """
+    version_to_replace = get_version_to_replace_with(version,file_path)
+    with open(file_path, "r") as f:
+        file_content = f.read()
+        python_version = (f"python_version = \"{version_to_replace}\"" 
+                          if "Pipfile" in file_path 
+                          else f"python = \"{version_to_replace}\"")
+        if full_str_python_version != python_version:
+                print(f"[INFO] change {file_path}")
+                file_content = file_content.replace(full_str_python_version,python_version)
+                with open(file_path, "w") as f:
+                    f.write(file_content)
+                return True
+        return False
+
+def change_pyproject_or_pipfile(file_path:str, str_version:str) -> Tuple[bool,str]:
+    """Replace the current python version in the Pipfile or the pyproject.toml.
+    
+        Args:
+            file_path (str): The file path to the Pipfile or the pyproject.toml file.
+            version (str): The version.
+            full_str_python_version (str): The older version.
+    """
+    current_version, full_str_python_version, success_extraction=extract_current_python_version(file_path)
+    if success_extraction:
+        result=replace_python_version(file_path, str_version, full_str_python_version)
+        return result,current_version
+    else:
+        print(f"[ERROR] can't extract python version form: {file_path}")
+    return False, ""
+
+def run_lock(base_path_docker:str,pipfile_or_pyproject_path:str)->bool:
+    """Runs poetry lock --no-update or pipfile lock --keep-outdated.
+    
+        Args:
+            base_path_docker (str): The DockerFile path.
+            pipfile_or_pyproject_path (str): The file path to the Pipfile or the pyproject.toml file.
+    """
+    base_path=base_path_docker.replace("/Dockerfile", "")
+    current_directory = os.getcwd()
+    os.chdir(f"{current_directory}/" + base_path)
+    try:
+        if "Pipfile" in pipfile_or_pyproject_path:
+            # waits for the process to end.
+            completed_process = subprocess.run(["pipenv", "lock", "--keep-outdated"], 
```

Comment:
are we printing the output/ err output here if process wasnt successful?

Reply:
Yes,  the line is: ` except subprocess.CalledProcessError as e:
        print(f"[ERROR] Lock failed with error: {str(e.stderr)} for {base_path_docker}")`